src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -395,7 +396,97 @@ struct KVAxesPosition {
     uint32_t batch;
     uint32_t seq_len;
 };
+} // anonymous namespace
 
+class CutTailAfterSlice : public ov::pass::MatcherPass {
+public:
+    OPENVINO_MATCHER_PASS_RTTI("npuw::patterns::CutTailAfterSlice");
+    CutTailAfterSlice::CutTailAfterSlice(
+        std::shared_ptr<std::pair<ov::OutputVector, ov::ParameterVector>> separated_tail_protocol) {
+        auto slice = opp::wrap_type<ov::op::v8::Slice>({opp::any_input(), opp::any_input(),
+                                                        opp::any_input(), opp::any_input(),
+                                                        opp::any_input()});
+        auto matmul = opp::wrap_type<ov::op::v0::MatMul>({slice, opp::any_input()});
+
+        // There are several patterns for matmul we are looking for:
+        // Matmul -> Result
+        // Matmul -> Add -> Result
+        auto matmul_add = opp::wrap_type<ov::op::v1::Add>({matmul, opp::any_input()});
+        // Matmul -> Transpose -> Result
+        auto matmul_transpose = opp::wrap_type<ov::op::v1::Transpose>({matmul, opp::any_input()});
+        // MatMul -> Divide -> Tanh -> Multiply -> Result
+        auto div = opp::wrap_type<ov::op::v1::Multiply, ov::op::v1::Divide>({matmul, opp::any_input()});
+        auto tanh = opp::wrap_type<ov::op::v0::Tanh>({div});
+        auto matmul_multiply = opp::wrap_type<ov::op::v1::Multiply>({tanh, opp::any_input()});
+
+        auto last_op = std::make_shared<ov::pass::pattern::op::Or>(
+            ov::OutputVector{matmul->output(0), matmul_add->output(0),
+                             matmul_transpose->output(0), matmul_multiply->output(0)});
+        auto res = opp::wrap_type<ov::op::v0::Result>({last_op->output(0)});
+
+        auto callback = [=](ov::pass::pattern::Matcher& m) {
+            auto& node_to_output = m.get_pattern_value_map();
+    
+            auto matched_node_slice = node_to_output.at(slice).get_node_shared_ptr();
+            auto matched_node_matmul = node_to_output.at(matmul).get_node_shared_ptr();
+            std::shared_ptr<ov::Node> matched_node_last_op = nullptr;
+            if (node_to_output.count(matmul_add)) {
+                matched_node_last_op = node_to_output.at(matmul_add).get_node_shared_ptr();
+            } else if (node_to_output.count(matmul_transpose)) {
+                matched_node_last_op = node_to_output.at(matmul_transpose).get_node_shared_ptr();
+            } else if (node_to_output.count(matmul_multiply)) {
+                matched_node_last_op = node_to_output.at(matmul_multiply).get_node_shared_ptr();
+            } else {
+                matched_node_last_op = matched_node_matmul;
+            }
```

Comment:
Would a simple `node_to_output.at(last_op)` work?
Also I doubt if you need this at all.

Reply:
It didn't work during development, but might be there was another issue, let me check